packages/kit/src/core/generate_manifest/index.js

```diff
@@ -71,8 +72,8 @@ export function generate_manifest(
 							params: ${get_params(route.params)},
 							path: ${route.path ? s(route.path) : null},
 							shadow: ${route.shadow ? importer(`${relative_path}/${build_data.server.vite_manifest[route.shadow].file}`) : null},
-							a: ${s(route.a.map(component => component && bundled_nodes.get(component).index))},
-							b: ${s(route.b.map(component => component && bundled_nodes.get(component).index))}
+							a: ${s(route.a.map(component => component && bundled_nodes.get(component)?.index))},
+							b: ${s(route.b.map(component => component && bundled_nodes.get(component)?.index))}
```

Comment:
why do we need `?.` all of a sudden?

Reply:
it's a result of typing `bundled_nodes`. as far as the typescript compiler knows getting a value from a map can always return `undefined`. it didn't type check it before when the type was `any`

Reply:
_We_ know that it's defined though, even if TypeScript doesn't. I'm not sure if there are any performance downsides to using `?.` instead of `.` but it feels like the correct thing to do is give TypeScript the information it needs rather than affecting runtime behaviour. One sec